mesonbuild/backend/xcodebackend.py

```diff
@@ -205,6 +205,9 @@ def __init__(self, build: T.Optional[build.Build], interpreter: T.Optional[Inter
         self.regen_dependency_id = self.gen_id()
         self.top_level_dict = PbxDict()
         self.generator_outputs = {}
+        self.arch = self.interpreter.builtin['target_machine'].cpu_method(None, None)
```

Comment:
target_machine indicates when building a compiler, which type of CPU the compiler emits code for. It's not the same as build_machine (the machine running meson) or host_machine (the cross-compile platform).

Is this correct? What's the goal here?

Reply:
This is not the way to do this either (I know the other backends do this, but they're wrong). This should be `self.build.machines.host.cpu` (or target, but probably host, as Eli points out)

Reply:
I have implemented that suggestion, but I was getting runtime errors, so it's now `self.build.environment.machines.host.cpu`.